### ISSUE TITLE
RCS1235 results into recursive call

### ISSUE THREAD

**Issue by nvmkpk** [JosefPihrt/Roslynator#1053]
**Product and Version Used**: 
Visual studio extension version 4.2.0

**Steps to Reproduce**:
Define code like below:
```csharp
class MyCollection : Collection<Class1>
{
    public void Add(Class2 item)
    {
        this.Add(item.ToClass1()));
    }

    public void AddRange(IEnumerable<Class2> data)
    {
        foreach (var item in data)
        {
            this.Add(item); // RCS1235 is reported here
        }
    }
}
```

**Actual Behavior**:
No RCS1235 should be reported (but does not report this on `RemoveRange` implementation.

**Expected Behavior**:
Below refactor is suggested but it results into recursive call.
```csharp
class MyCollection : Collection<Class1>
{
    public void Add(Class2 item)
    {
        this.Add(item.ToClass1()));
    }

    public void AddRange(IEnumerable<Class2> data)
    {
        this.AddRange(data); // Recursive call
    }
}
```
